### PR TITLE
Create search indexes for Answer content

### PR DESCRIPTION
**Pull request by higs4281**
This PR introduces two Elasticsearch indexes, for English and Spanish
content, based on a unified Answer model, plus URLs and views for
searching and displaying results. Our goal is to handle category and
audience faceting client-side, so that search can focus on finding
answers by content.

Migration note: This PR squashes ask_cfpb migrations in preparation for
exposing the models to databases other than build, so if you had pulled
in migrations from previous ask_cfpb pull requests, you might want to
clean out your `.pyc` files with `python cfgov/manage.py clean_pyc -p .`

## Testing
You'll need to have Elasticsearch 1.x running (to match the version on our servers) and do the migration dance, with an index command added:

```
export DEPLOY_ENVIRONMENT=build
./drop-db.sh
./create-mysql-db.sh
mysql -uroot v1 < ~/Downloads/production_django.sql
python cfgov/manage.py migrate
python cfgov/manage.py runscript migrate_knowledgebase
python cfgov/manage.py update_index ask_cfpb
./runserver
```

If you set your Wagtail default Site to `localhost` and port 8000, you can see search results:   

English:
- http://localhost:8000/ask-cfpb/search-en/?q=college%20loan

As json:
- http://localhost:8000/ask-cfpb/search-en/json/?q=college%20loan

And Spanish:
- http://localhost:8000/ask-cfpb/search-es/?q=tarjeta%20respuestas

## Front-end note
The html files added here are stubs intended for testing search results, and are not even close to being the real thing.

### CODE REVIEW

**Comment by willbarton** [comment]
Musing while reviewing: I wonder if the squashed migrations will cause any deployment problems on build.

File: cfgov/ask_cfpb/views.py [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
 
-from .models.pages import AnswerPage
+# from haystack.generic_views import SearchView
+from haystack.query import SearchQuerySet
+from haystack.inputs import Clean
+
+from django.shortcuts import get_object_or_404, redirect  # render_to_response
+from django.http import HttpResponse
+
+from .models import (
```
Comment (willbarton) [inline cfgov/ask_cfpb/views.py]: Tiny nit I'm going to pick: can you use an absolute import?

**Comment by higs4281** [comment]
> Musing while reviewing: I wonder if the squashed migrations will cause any deployment problems on build.

It certainly will. I will manually fix the django_migrations table and ask_cfpb tables in the build db. Then we'll be ready for a fresh start on other servers.

File: cfgov/ask_cfpb/views.py (hunk 1 above)
Comment (higs4281) [inline cfgov/ask_cfpb/views.py]: The curse of Two Scoops

**Pull request merged**